# Task 1: Load and Preview Dataset

In [11]:
import pandas as pd

url = "https://drive.google.com/uc?id=1Mlu9r3afSGz5_02Y3wDJoQLpXszQXaIU"

df = pd.read_csv(url, index_col="order_id")

df.head()

,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1001,Alice Johnson,Laptop,1.0,85000.0,2024-01-15,delivered
1002,Bob Smith,Mouse,3.0,499.0,2024-01-16,pending
1003,Carol White,Monitor,2.0,15000.0,NaN,delivered
1004,Dave Brown,Keyboard,NaN,1200.0,2024-01-18,cancelled
1005,Eva Green,Laptop,1.0,85000.0,2024-01-19,delivered


In [12]:
df.tail()

,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1016,Paul Harris,Mouse,6.0,499.0,2024-01-30,pending
1017,Quinn Martin,Monitor,NaN,14500.0,2024-01-31,delivered
1018,Rachel King,Laptop,1.0,85000.0,NaN,delivered
1019,Sam Wright,Keyboard,3.0,1200.0,2024-02-01,cancelled
1020,Tina Baker,USB Hub,2.0,799.0,2024-02-02,pending


# Task 2: Structural Inspection

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   customer_name  20 non-null     object 
 1   product        20 non-null     object 
 2   quantity       16 non-null     float64
 3   unit_price     20 non-null     float64
 4   order_date     16 non-null     object 
 5   status         20 non-null     object 
dtypes: float64(2), object(4)
memory usage: 1.1+ KB


In [14]:
df.shape

(20, 6)

In [15]:
df.isnull().sum()

,0
customer_name,0
product,0
quantity,4
unit_price,0
order_date,4
status,0


In [16]:
df["order_date"] = pd.to_datetime(df["order_date"])

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   customer_name  20 non-null     object        
 1   product        20 non-null     object        
 2   quantity       16 non-null     float64       
 3   unit_price     20 non-null     float64       
 4   order_date     16 non-null     datetime64[ns]
 5   status         20 non-null     object        
dtypes: datetime64[ns](1), float64(2), object(3)
memory usage: 1.1+ KB


## Task 2 Observations

- The dataset has 20 rows and 6 columns.
- The columns with missing values are:
  - quantity (4 null values)
  - order_date (4 null values)
- The order_date column required datatype conversion.
- I used pd.to_datetime() function for conversion.

# Task 3: Statistical Summary

In [18]:
df.describe()

,quantity,unit_price,order_date
count,16.000000,20.000000,16
mean,2.437500,20224.600000,2024-01-24 09:00:00
min,1.000000,499.000000,2024-01-15 00:00:00
25%,1.000000,799.000000,2024-01-19 18:00:00
50%,2.000000,2500.000000,2024-01-24 12:00:00
75%,3.000000,14625.000000,2024-01-29 06:00:00
max,6.000000,85000.000000,2024-02-02 00:00:00
std,1.504161,33573.389607,NaN


## Task 3 Observations

- The median unit_price is ₹2500.
- The minimum unit_price is ₹499.
- The maximum unit_price is ₹85000.
- The unit_price distribution appears positively skewed
  because the maximum value is much higher than the median.

The unit_price appears to be positively skewed (right-skewed).
This is because the maximum value (85000) is much larger than the median (2500),
indicating the presence of high-value outliers pulling the distribution to the right.